code/src/reducers/todos.js

```diff
@@ -0,0 +1,62 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const todos = createSlice({
+    name:'todos',
+    initialState: {
+        items: [
+            { id: 1, description: 'Groceries🍎', isComplete: false },
+            { id: 2, description: 'Coding💻', isComplete: true },
+            { id: 3, description: 'Housekeeping🏠', isComplete: false },
+            { id: 4, description: 'Workout🤸‍♀️', isComplete: false }
+        ],
+
+    },
+    reducers: {
+        toggleComplete: (store, action) => {
+            //v1 - immutability
+            const updatedItems = store.items.map(todo => {
+                if(todo.id === action.payload) {
+                    //update iscOMPLETE property
+                    return {
+                        ...todo,
+                        isComplete: !todo.isComplete
+                    }
+                } else {
+                    return todo;
+                    //leave it as it was
+                }
+            });
+            //this store here is the main difference between mutability and immutability
+            // (this one is only valid for immutability)
+            store.items = updatedItems;
+           //v2 mutability
+            //store.items.forEach(todo => {
+               // if (todo.id === action.payload) {
+                    //  todo.isComplete = !todo.isComplete;
+                //}
+           // })
+        },
+        removeTodo: (store, action) => {
+            // v1 - Immutability
+            // const decreasedItems = store.items.filter(todo => todo.id !== action.payload);
+
+            // store.items = decreasedItems;
+
+            //v2 - Mutability
+            store.items.splice(action.payload, 1);
```

Comment:
you are passing the id as the payload, not the index, that is why your reducer removes the wrong item from the array.

You have 3 options here(from best to worst):
1. use the Immutability version because it looks good for me,
2. find index of task with id from the payload, something like this:
```jsx
const index = store.map(e => e.id).indexOf(action.payload);
store.items.splice(index, 1);
```
3: pass index as payload.